src/hb-ot-cmap-table.hh

```diff
@@ -1441,17 +1446,47 @@ struct cmap
 
     const EncodingRecord *unicode_bmp= nullptr, *unicode_ucs4 = nullptr, *ms_bmp = nullptr, *ms_ucs4 = nullptr;
     bool has_format12 = false;
+    hb_hashmap_t<unsigned, const EncodingRecord *, (unsigned)-1, nullptr> unicode_bmp_tables;
+    hb_hashmap_t<unsigned, const EncodingRecord *, (unsigned)-1, nullptr> ms_bmp_tables;
+    hb_map_t format12_tables;
 
-    for (const EncodingRecord& _ : encodingrec_iter)
+    for (auto _ : encodingrec_iter)
```

Comment:
This for loop, and down to line 1493 could become a separate method.

It would take encoding iter and the three maps as output parameters and return a boolean, which returns the result of the checks on lines 1492 and 1493.

Name could be something like, "categorize_sub_tables"